src/sql/src/plan/lowering.rs

```diff
@@ -736,11 +736,17 @@ impl HirRelationExpr {
                                     config.enable_new_outer_join_lowering,
                                     config.enable_outer_join_null_filter,
                                 )? {
+                                    metrics.iter().for_each(|metrics| {
```

Comment:
`metrics` is an `Option`. Is there a reason why you don't use an `if let` here?

(Also applies to the other instances of the same pattern below.)